conda-store-server/conda_store_server/api.py

```diff
@@ -20,11 +21,21 @@ def get_namespace(db, name: str = None, id: int = None):
     return db.query(orm.Namespace).filter(*filters).first()
 
 
-def list_environments(db, namespace: str = None, search=None):
+def list_environments(db, namespace: str = None, search: str = None):
     filters = []
     if namespace:
         filters.append(orm.Namespace.name == namespace)
 
+    if search:
+        new_filter = or_(
+            orm.Environment.name.like(f"{search}"),
+            orm.Environment.name.like(f"%{search}"),
+            orm.Environment.name.like(f"{search}%"),
+            orm.Environment.name.like(f"%{search}%"),
```

Comment:
Is there a reason that all four like queries must be run? Would the %search% not match all of the above statements? Could definitely bey misunderstanding of sql like queries

Reply:
I just checked and % is zero or more. While _ is one or more.

Reply:
When I tested with only `orm.Environment.name.like(f"%{search}%")` as filter, I had some wrong results in edge cases (like searching for "data_" and not retrieving "data_engineering", for instance). This led me to this `or_` thing. But I may have tested wrong as I realize, after testing again, that only `%search%` alone works perfectly.

this inspires me for another PR with solid unit tests over the API.

Anyway, I have commited a simpler version

Reply:
We'll need to sanitize the query. `_` and `%` are reserved for wildcard matches. Also would be nice if we could ensure that the query matches a regex so that we can't have sql injection. But this does look like a great PR!

Reply:
I have pushed a much better version in my last commit.
Instead of using `.like()`, let's use `.contains()` with the `autoescape=True` parameter.

More details in the [SQLAlchemy docs](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.contains)